src/nbl/asset/interchange/CIESProfileLoader.h

```diff
@@ -0,0 +1,160 @@
+
+private:
+  PhotometricType type;
+  core::vector<double> hAngles;
+  core::vector<double> vAngles;
+  core::vector<double> data;
+};
+
+class CIESProfileParser {
+public:
+  CIESProfileParser(char *buf, size_t size) { ss << std::string(buf, size); }
+
+  bool parse(CIESProfile& result);
+
+private:
+  int getInt(const char *errorMsg) {
+    int in;
+    if (ss >> in)
+      return in;
+    error = true;
+    if (!this->errorMsg)
+      this->errorMsg = errorMsg;
+    return 0;
+  }
+
+  double getDouble(const char *errorMsg) {
+    double in;
+    if (ss >> in)
+      return in;
+    error = true;
+    if (!this->errorMsg)
+      this->errorMsg = errorMsg;
+    return -1.0;
+  }
+
+  bool error{false};
+  const char *errorMsg{nullptr};
+  std::stringstream ss;
+};
+
+class CIESProfileMetadata final : public asset::IAssetMetadata {
+public:
+  CIESProfileMetadata(double maxIntensity)
+      : IAssetMetadata(), maxIntensity(maxIntensity) {}
+
+  _NBL_STATIC_INLINE_CONSTEXPR const char *LoaderName = "CIESProfileLoader";
+  const char *getLoaderName() const override { return LoaderName; }
+
+  double getMaxIntensity() const { return maxIntensity; }
+
+private:
+  double maxIntensity;
+};
+
+class CIESProfileLoader final : public asset::IAssetLoader {
+public:
+  _NBL_STATIC_INLINE_CONSTEXPR size_t TEXTURE_WIDTH = 1024;
+  _NBL_STATIC_INLINE_CONSTEXPR size_t TEXTURE_HEIGHT = 2048;
+  _NBL_STATIC_INLINE_CONSTEXPR double MAX_VANGLE = 180.0;
+  _NBL_STATIC_INLINE_CONSTEXPR double MAX_HANGLE = 360.0;
+
+  //! Check if the file might be loaded by this class
+  /** Check might look into the file.
+  \param file File handle to check.
+  \return True if file seems to be loadable. */
+  bool isALoadableFileFormat(io::IReadFile *_file) const override {
+    const size_t begginingOfFile = _file->getPos();
+    _file->seek(0ull);
+    std::string versionBuffer(5, ' ');
+    _file->read(versionBuffer.data(), versionBuffer.size());
+    _file->seek(begginingOfFile);
+    return versionBuffer == "IESNA";
+  }
+
+  //! Returns an array of string literals terminated by nullptr
+  const char **getAssociatedFileExtensions() const override {
+    static const char *extensions[]{"ies", nullptr};
+    return extensions;
+  }
+
+  //! Returns the assets loaded by the loader
+  /** Bits of the returned value correspond to each IAsset::E_TYPE
+  enumeration member, and the return value cannot be 0. */
+  uint64_t getSupportedAssetTypesBitfield() const override {
+    return asset::IAsset::ET_IMAGE;
+  }
+
+  //! Loads an asset from an opened file, returns nullptr in case of failure.
+  asset::SAssetBundle
+      loadAsset(io::IReadFile* _file,
+          const asset::IAssetLoader::SAssetLoadParams& _params,
+          asset::IAssetLoader::IAssetLoaderOverride* _override = nullptr,
+          uint32_t _hierarchyLevel = 0u) override;
+
+private:
+    core::smart_refctd_ptr<asset::ICPUImage>
+        createTexture(const CIESProfile& profile, size_t width, size_t height);
```

Comment:
IMHO you should load and return an ICPUImageView, since you know exactly:
- usage (sampled)
- final view format (R16_UNORM)
- the colorspace (identity)
- whether the sampler is meant to be 2D or 2D_ARRAY

which is not something we can say about the other "image loaders" like PNG, JPG, EXR